app/lib/backend_api_logic/metric_extension.rb

```diff
@@ -32,18 +32,26 @@ def hits_extended_system_name_as_sql
       def hits_extended_system_name_regex
         /\Ahits(#{Regexp.escape(SYSTEM_NAME_SUFFIX_SEPARATOR)}\d+)?\z/
       end
+
+      def build_extended_system_name(system_name, owner_id:)
```

Comment:
This probably should have a unit test of its own 😄

Reply:
Done.